### PR TITLE
language: dont change directory for generation of ifaces

### PR DESCRIPTION
**Pull request by robin-da**
We don't need to change the directory to generate the interface files.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/lib/DA/Cli/Damlc.hs [hunk 1]
```diff
@@ -638,33 +636,38 @@ createProjectPackageDb opts fps = do
...
-        mapM_ writeSrc src
+        let src' = [ (toNormalizedFilePath $ workDir </> fromNormalizedFilePath nfp, str) | (nfp, str) <- src]
+        mapM_ writeSrc src'
         opts' <-
             mkOptions $
             opts
-                { optWriteInterface = True
-                , optPackageDbs = projectPackageDatabaseAbs : optPackageDbs opts
-                , optIfaceDir = Just "./"
+                { optWriteInterface = False
```
Comment (cocreature) [inline compiler/damlc/lib/DA/Cli/Damlc.hs]: Why did this change?

File: compiler/damlc/lib/DA/Cli/Damlc.hs [hunk 2]
```diff
@@ -638,33 +636,38 @@ createProjectPackageDb opts fps = do
...
                 , optIsGenerated = True
                 , optDflagCheck = False
                 , optMbPackageName = Just unitIdStr
                 , optHideAllPkgs = False
                 , optGhcCustomOpts = []
                 , optPackageImports = []
+                , optImportPath = workDir : optImportPath opts
                 }
+
+        putStrLn $ show $ optImportPath opts'
```
Comment (cocreature) [inline compiler/damlc/lib/DA/Cli/Damlc.hs]: Looks like leftover debugging output.

File: compiler/damlc/lib/DA/Cli/Damlc.hs [hunk 3]
```diff
@@ -638,33 +636,38 @@ createProjectPackageDb opts fps = do
...
+        mapM_ writeSrc src'
         opts' <-
             mkOptions $
             opts
-                { optWriteInterface = True
-                , optPackageDbs = projectPackageDatabaseAbs : optPackageDbs opts
-                , optIfaceDir = Just "./"
+                { optWriteInterface = False
+                , optPackageDbs = projectPackageDatabase : optPackageDbs opts
+                , optIfaceDir = Nothing
```
Comment (cocreature) [inline compiler/damlc/lib/DA/Cli/Damlc.hs]: Same question here, do we need to iface dir at all now?

File: compiler/damlc/lib/DA/Cli/Damlc.hs [hunk 4]
```diff
@@ -597,20 +597,18 @@ createProjectPackageDb opts fps = do
...
+                          dbPath
+                          projectPackageDatabase
+                          unitIdStr
+                          pkgIdStr
+                          pkgName
+                          mbPkgVersion
+                          deps
 
                       unless (null templInstSrc) $
                           generateAndInstallInstancesPkg
```
Comment (cocreature) [inline compiler/damlc/lib/DA/Cli/Damlc.hs]: We do basically the same thing in `generateAndInstallInstancesPkg` so we should probably fix that one as well (fine to leave that for a separate PR).

File: compiler/damlc/lib/DA/Cli/Damlc.hs (hunk 1 above)
Comment (robin-da) [reply compiler/damlc/lib/DA/Cli/Damlc.hs]: We write the interfaces with `writeIfaceAndHie` in the next step. This way we don't do it two times.

File: compiler/damlc/lib/DA/Cli/Damlc.hs (hunk 2 above)
Comment (robin-da) [reply compiler/damlc/lib/DA/Cli/Damlc.hs]: will remove.

File: compiler/damlc/lib/DA/Cli/Damlc.hs (hunk 3 above)
Comment (robin-da) [reply compiler/damlc/lib/DA/Cli/Damlc.hs]: same as above.

File: compiler/damlc/lib/DA/Cli/Damlc.hs (hunk 1 above)
Comment (cocreature) [reply compiler/damlc/lib/DA/Cli/Damlc.hs]: So this was redundant before this PR as well? Makes sense in that case, thanks for the explanation!

Comment (robin-da) [reply compiler/damlc/lib/DA/Cli/Damlc.hs]: yes, this was redundant before.
